### PR TITLE
Add test 'CRUD tag'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add test 'CRUD tag'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/07_search/02_tag/01_CRUDTag.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21636)

### CODE REVIEW

**Review by SimonGrn — DISMISSED** [review]

File: tests/UI/pages/BO/shopParameters/search/tags/index.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  async filterTable(page, filterType, filterBy, value = '') {
+    switch (filterType) {
+      case 'input':
+        await this.setValue(page, this.filterColumn(filterBy), value);
+        break;
+      case 'select':
+        await this.selectByVisibleText(page, this.filterColumn(filterBy), value);
+        break;
+      default:
+      // Do nothing
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/search/tags/index.js]:
```suggestion
      throw new Error(`${filterBy} was not found as a filter`);
```

File: tests/UI/campaigns/data/faker/tag.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const {Products} = require('@data/demo/products');
+const {Languages} = require('@data/demo/languages');
+
+const productsNames = Object.values(Products).map(product => product.name);
+const LanguagesNames = Object.values(Languages).map(language => language.name);
+
+module.exports = class Supplier {
+  constructor(tagsToCreate = {}) {
+    this.name = tagsToCreate.name || `new_tag_${faker.lorem.word()}`;
+    this.language = tagsToCreate.language || faker.random.arrayElement(LanguagesNames);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/data/faker/tag.js]:
```suggestion
const productsNames = Object.values(Products).map(product => product.name);
const languagesNames = Object.values(Languages).map(language => language.name);

module.exports = class Supplier {
  constructor(tagsToCreate = {}) {
    this.name = tagsToCreate.name || `new_tag_${faker.lorem.word()}`;
    this.language = tagsToCreate.language || faker.random.arrayElement(languagesNames);
```

Comment (nesrineabdmouleh) [reply tests/UI/campaigns/data/faker/tag.js]: Done thx :+1:

File: tests/UI/pages/BO/shopParameters/search/tags/index.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/shopParameters/search/tags/index.js]: Done :+1:

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/campaigns/functional/BO/13_shopParameters/07_search/02_tag/01_CRUDTag.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+const addTagPage = require('@pages/BO/shopParameters/search/tags/add');
+
+// Import data
+const TagFaker = require('@data/faker/tag');
+const {Languages} = require('@data/demo/languages');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_shopParameters_tags_CRUDTag';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/13_shopParameters/07_search/02_tag/01_CRUDTag.js]:
```suggestion
const baseContext = 'functional_BO_shopParameters_search_tag_CRUDTag';
```

**Review by Progi1984 — DISMISSED** [review]

File: tests/UI/campaigns/functional/BO/13_shopParameters/07_search/02_tag/01_CRUDTag.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/13_shopParameters/07_search/02_tag/01_CRUDTag.js]: Done thx :+1:

**Review by boubkerbribri — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**